client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -377,6 +378,12 @@ function mediaButton( editor ) {
 				editor.fire( 'SetTextAreaContent', { content } );
 			}
 
+			// We have just replaced some image nodes, potentially losing selection/focus.
+			// So if an image node was selected and one isn't now, re-select it.
```

Comment:
The duplication bug occurred because an edited image was getting deselected, then modified, and thusly appended rather than replaced.

This change fixes the cause of that issue: the deselection.